contracts/access/manager/AccessManager.sol

```diff
@@ -572,15 +572,13 @@ contract AccessManager is Context, Multicall, IAccessManager {
 
         uint48 minWhen = Time.timestamp() + setback;
 
-        if (when == 0) {
-            when = minWhen;
-        }
-
-        // If caller is not authorised, revert
-        if (!immediate && (setback == 0 || when < minWhen)) {
+        // if call is not authorized, or if requested timing is sooner than
+        if ((!immediate && setback == 0) || (when > 0 && when < minWhen)) {
```

Comment:
Made this for comprehension. Sharing if it helps anyone reviewing

| !immediate | setback == 0 | when > 0 | when < minWhen | ((!immediate && setback == 0) OR (when > 0 && when < minWhen)) | Interpretation |
|------------|--------------|----------|----------------|----------------------------------------------------------------| -------------- |
| 0          | 0            | 0        | 0              | PASS    | Authorized with setback, "when" is 0 and minWhen is also 0 (minWhen can't be 0) 
| 0          | 0            | 0        | 1              | PASS    | Authorized with setback, "when" is 0 and minWhen is not 0
| 0          | 0            | 1        | 0              | PASS    | Authorized with setback, "when" is not 0 and larger than minWhen
| 0          | 0            | 1        | 1              | REVERT  | Authorized with setback, "when" is not 0 but smaller than minWhen
| 0          | 1            | 0        | 0              | PASS    | Authorized with no setback, "when" is 0 and minWhen is also 0 (minWhen can't be 0)
| 0          | 1            | 0        | 1              | PASS    | Authorized with no setback, "when" is 0 and minWhen is not 0
| 0          | 1            | 1        | 0              | PASS    | Authorized with no setback, "when" is not 0 and larger than minWhen
| 0          | 1            | 1        | 1              | REVERT  | Authorized with no setback, "when" is not 0 but smaller than minWhen
| 1          | 0            | 0        | 0              | PASS    | Not authorized with setback, "when" is 0 and minWhen is also 0 (minWhen can't be 0)
| 1          | 0            | 0        | 1              | PASS    | Not authorized with setback, "when" is 0 and minWhen is not 0
| 1          | 0            | 1        | 0              | PASS    | Not authorized with setback, "when" is not 0 and larger than minWhen
| 1          | 0            | 1        | 1              | REVERT  | Not authorized with setback, "when" is not 0 but smaller than minWhen
| 1          | 1            | 0        | 0              | REVERT  | Not authorized with no setback, "when" is 0 and minWhen is also 0 (minWhen can't be 0)
| 1          | 1            | 0        | 1              | REVERT  | Not authorized with no setback, "when" is 0 and minWhen is not 0
| 1          | 1            | 1        | 0              | REVERT  | Not authorized with no setback, "when" is not 0 and larger than minWhen
| 1          | 1            | 1        | 1              | REVERT  | Not authorized with no setback, "when" is not 0 but smaller than minWhen